### DISCUSSION TITLE
missing embedded fonts

### DISCUSSION THREAD

**Discussion by devKyanon** [koreader/koreader#11289] [Q&A]
Hello!

I have a question about fonts.

Some of my epub files definitely have embedded fonts and its css denotes it well with defining font families.
However, koreader sometimes does not recognize it so that fonts are not applied in the text. 
(I mean, the fonts are recognized well for some epubs (they work fine for other ebook apps)).

I have tried to solve this problem with applying user style tweaks but I cannot find the correct solution.

Could you help me?

**Comment by mergen3107** [comment]
Bottom menu - last tab - Embedded fonts?

**Reply by devKyanon** [reply]
Of course, I have checked it.
But the button for the embedded font on/off is not activated.

I can only turn on/off the embedded style.

**Reply by mergen3107** [reply]
That probably means either KOReader didn't like the fonts and threw them out :D or that they are defined in the css, but fonts were stripped.

If want to investigate this further, you need to gather verbose logs and provide a scrambled epub.

Or just open the epub from PC as an archive and look inside for the fonts.

**Reply by devKyanon** [reply]
Yeah, I explicitly put the required font files (I extracted it from the epub) into /Fonts but still it doesn't work.
On other epub reader apps (such as moon reader) or on my PC when I open the html files in the epub directly, the fonts are clearly shown. 

I don't know why only koreader cannot get the font correctly...:(
Once this is fixed, koreader is really the best reader app for me.

(I guess, the fonts are for korean, so it might be the unicode issue).

Then, how can I get the logs?

**Reply by NiLuJe** [reply]
We'd probably need access to an affected book. Unfortunately, the scramble plugin doesn't support CJK, so, here be (legal) dragons ;).
